### PR TITLE
Add release workflow

### PR DESCRIPTION
**Pull request by maruu**

### CODE REVIEW

File: .github/workflows/gen-release-pr.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+    - name: Batch and merge
+      run: |
+        changie batch auto
+        changie merge
+        echo RELEASE_VERSION=$(changie latest) >> $GITHUB_ENV
+    - name: Create Pull Request
```
Comment (maruu) [inline .github/workflows/gen-release-pr.yml:19]: I saw this in the changie repository. I am wondering why this does not need a github token?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+    - name: Add env vars
+      run: |
+        echo GORELEASER_CURRENT_TAG=$(changie latest) >> $GITHUB_ENV
+        echo RELEASE_NOTES_PATH=.changes/$(changie latest).md >> $GITHUB_ENV
+    - name: Create release
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (maruu) [inline .github/workflows/release.yml:24]: not sure if this is required or already implicitly set?

File: .github/workflows/gen-release-pr.yml (hunk 1 above)
Comment (codetent) [reply .github/workflows/gen-release-pr.yml:19]: I guess this is possible using the integrated workflow token. Not sure if the correct rights are set for it. But let's test....

File: .github/workflows/release.yml (hunk 2 above)
Comment (codetent) [reply .github/workflows/release.yml:24]: I guess this is needed

File: .github/workflows/gen-release-pr.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+      uses: peter-evans/create-pull-request@v5
+      with:
+        title: Release ${{ env.RELEASE_VERSION }}
+        branch: release/${{ env.RELEASE_VERSION }}
+        commit-message: Release ${{ env.RELEASE_VERSION }}
```
Comment (codetent) [inline .github/workflows/gen-release-pr.yml:24]:
```suggestion
        commit-message: prepare ${{ env.RELEASE_VERSION }} release notes
```

**Review by codetent — APPROVED** [review]

**Pull request merged**